daemon/containerio/cio.go

```diff
@@ -116,7 +116,7 @@ func copyIO(fifos *containerdio.FIFOSet, ioset *ioSet, tty bool) (_ *wgCloser, e
 		}
 	}()
 
-	if fifos.In != "" {
+	if fifos.In != "" && ioset.in.(*ContainerIO) != nil {
```

Comment:
Use `reflect` maybe a good choice. 

However, I think we can change the view to this bug. RestoreContainer use `WithAttach` to load the task. The `fifo` path does exist so that the `fifos.In` doesn't equal to empty. From this view, I think we can set the `path.In` to empty. In my opinion, I don't think we can restore the container with the stdin. The restored container should be detached. WDYT?

Reply:
I agree with you. Did you mean to set the `paths.In` in `WithAttach` to empty? @fuweid